### PR TITLE
- Changed the route for the note editor screen

### PR DESCRIPTION
**Pull request by sadatetchamouza**
- Defined note as the list of note from à specific book
- Updated every lib in the pubspec.yaml
- Changed ML vision lib to the new one
- Updated note_screen.dart to be the book detail so thath we can display book's notes

### CODE REVIEW

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -35,6 +33,10 @@
 	<string>$(FLUTTER_BUILD_NUMBER)</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<key>NSCameraUsageDescription</key>
+	<string>Can I use the camera please?</string>
+	<key>NSMicrophoneUsageDescription</key>
+    <string>Can I use the mic please?</string>
```
Comment (rygelouv) [inline ios/Runner/Info.plist:39]: Can't figure out what these strings are for. Are they displayed in the app? We might need to update them with a better content

File: pubspec.yaml [hunk 2]
```diff
@@ -33,15 +33,15 @@ dependencies:
...
 
-  tuple: ^1.0.3
+  tuple: ^2.0.0
   intl: ^0.16.1
   url_launcher: ^5.4.2
   path_provider: ^1.6.7
-  firebase_ml_vision: ^0.9.4
-  camera: ^0.5.8+2
+  google_ml_kit: ^0.7.0
+  camera: ^0.8.1+7
```
Comment (rygelouv) [inline pubspec.yaml:44]: Good to see these libraries being updated.

File: lib/screen/note_screen.dart [hunk 3]
```diff
@@ -296,7 +292,7 @@ class _NoteScreen extends State<NoteScreen> with CommandHandler {
   /// Callback on a single note clicked
   void _onNoteTap(Note note) async {
     final command =
-    await Navigator.pushNamed(context, '/note', arguments: {'note': note});
+    await Navigator.pushNamed(context, '/note_editor', arguments: {'note': note});
```
Comment (rygelouv) [inline lib/screen/note_screen.dart:295]: So we're basically navigating to the node editor when the note is clicked?

**Review by rygelouv — APPROVED** [review]
Can't have it work as expected on my device. I can show what I have on Friday

File: ios/Runner/Info.plist (hunk 1 above)
Comment (sadatetchamouza) [reply ios/Runner/Info.plist:39]: No theses strings are for iOS configuration, basically these strings show up when we request the user permission to use his cam or mic

File: lib/screen/note_screen.dart (hunk 3 above)
Comment (sadatetchamouza) [reply lib/screen/note_screen.dart:295]: Yes exactly, I just rename the route to note_editor. The route name we had there was note but redirect to the editor, so I changed to an explicit name

**Pull request merged**